### ISSUE TITLE
maDLC colab gets stuck

### ISSUE THREAD

**Issue by roaldarbol** [DeepLabCut/DeepLabCut#1456]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Whilst trying out the new 2.2 version with the provided .ipynb, I enter what appears to be an infinity-loop during `train_network`. It seems that the relevant error message is `AttributeError: 'int' object has no attribute 'nb_rows'`. 

### Operating System

MacOS Big Sur 11.4

### DeepLabCut version

2.2

### DeepLabCut mode

multi animal

### Device type

Google COLAB GPU

### Steps To Reproduce

1. Create a new conda environment from the provided .yaml file
2. Create new project and label frames
3. Follow the .ipynb instructions, mounting drive and creating training data.
4. When running `train_network`, it gets stuck (see log).

### Relevant log output

```shell
Config:
{'all_joints': [[0], [1]],
 'all_joints_names': ['front', 'back'],
 'alpha_r': 0.02,
 'apply_prob': 0.5,
 'batch_size': 8,
 'clahe': True,
 'claheratio': 0.1,
 'crop_pad': 0,
 'crop_sampling': 'hybrid',
 'crop_size': [400, 400],
 'cropratio': 0.4,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_testAug9/test_me95shuffle11.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'decay_steps': 30000,
 'deterministic': False,
 'display_iters': 500,
 'edge': False,
 'emboss': {'alpha': [0.0, 1.0], 'embossratio': 0.1, 'strength': [0.5, 1.5]},
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'histeq': True,
 'histeqratio': 0.1, 
 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'lr_init': 0.0005,
 'max_input_size': 1500,
 'max_shift': 0.4,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_testAug9/Documentation_data-test_95shuffle11.pickle',
 'min_input_size': 64,
 'mirror': False,
 'multi_stage': True,
 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]],
 'net_type': 'resnet_50',
 'num_idchannel': 0,
 'num_joints': 2,
 'num_limbs': 1,
 'optimizer': 'adam',
 'pafwidth': 20,
 'pairwise_huber_loss': False,
 'pairwise_loss_weight': 0.1,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1]],
 'partaffinityfield_predict': True,
 'pos_dist_thresh': 17,
 'pre_resize': [],
 'project_path': '/content/drive/My Drive/DLC/test-me-2021-08-09',
 'regularize': False,
 'rotation': 25,
 'rotratio': 0.4,
 'save_iters': 10000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'sharpen': False,
 'sharpenratio': 0.3,
 'shuffle': True,
 'snapshot_prefix': '/content/drive/My '
                    'Drive/DLC/test-me-2021-08-09/dlc-models/iteration-0/testAug9-trainset95shuffle11/train/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}

Selecting multi-animal trainer
Activating limb prediction...
Batch Size is 8
Getting specs multi-animal-imgaug 1 2

/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/engine/base_layer_v1.py:1692: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '

Loading ImageNet-pretrained resnet_50
Max_iters overwritten as 75000
Display_iters overwritten as 100
Save_iters overwritten as 1000
Training parameters:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': '/content/drive/My Drive/DLC/test-me-2021-08-09/dlc-models/iteration-0/testAug9-trainset95shuffle11/train/snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'adam', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 8, 'dataset_type': 'multi-animal-imgaug', 'deterministic': False, 'mirror': False, 'pairwise_huber_loss': False, 'weigh_only_present_joints': False, 'partaffinityfield_predict': True, 'pairwise_predict': True, 'all_joints': [[0], [1]], 'all_joints_names': ['front', 'back'], 'alpha_r': 0.02, 'apply_prob': 0.5, 'clahe': True, 'claheratio': 0.1, 'crop_sampling': 'hybrid', 'crop_size': [400, 400], 'cropratio': 0.4, 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_testAug9/test_me95shuffle11.pickle', 'decay_steps': 30000, 'display_iters': 500, 'edge': False, 'emboss': {'alpha': [0.0, 1.0], 'embossratio': 0.1, 'strength': [0.5, 1.5]}, 'histeq': True, 'histeqratio': 0.1, 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt', 'lr_init': 0.0005, 'max_input_size': 1500, 'max_shift': 0.4, 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_testAug9/Documentation_data-test_95shuffle11.pickle', 'min_input_size': 64, 'multi_stage': True, 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]], 'net_type': 'resnet_50', 'num_idchannel': 0, 'num_joints': 2, 'num_limbs': 1, 'pafwidth': 20, 'pairwise_loss_weight': 0.1, 'partaffinityfield_graph': [[0, 1]], 'pos_dist_thresh': 17, 'pre_resize': [], 'project_path': '/content/drive/My Drive/DLC/test-me-2021-08-09', 'rotation': 25, 'rotratio': 0.4, 'save_iters': 10000, 'scale_jitter_lo': 0.5, 'scale_jitter_up': 1.25, 'sharpen': False, 'sharpenratio': 0.3}
Starting multi-animal training....

Exception in thread Thread-11:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/core/train.py", line 80, in load_and_enqueue
    batch_np = dataset.next_batch()
  File "/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/datasets/pose_multianimal_imgaug.py", line 327, in next_batch
    images=batch_images, keypoints=batch_joints
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/meta.py", line 1888, in __call__
    return self.augment(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/meta.py", line 1859, in augment
    batch_aug = self.augment_batch(batch, hooks=hooks)
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/meta.py", line 439, in augment_batch
    attr, hooks=hooks)
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/meta.py", line 603, in augment_images
    hooks=hooks
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/meta.py", line 2823, in _augment_images
    hooks=hooks
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/meta.py", line 515, in augment_images
    hooks=hooks
  File "/usr/local/lib/python3.7/dist-packages/imgaug/augmenters/size.py", line 1652, in _augment_images
    offset_xs, offset_ys = self._draw_samples(nb_images, random_state)
  File "/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/datasets/augmentation.py", line 52, in _draw_samples
    n_samples = batch.nb_rows
AttributeError: 'int' object has no attribute 'nb_rows'
```

### Anything else?
I have had some trouble with multiple Anaconda installations which have caused me some issues not too long ago, but if I understand correctly that shouldn't affect the colab, right?
And like always, can never be said enough - really love your work, big kudos to the entire team; hope you're enjoying some well-deserved time off at the moment! 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by akesner1** [comment]
I am having the same issue I think. At least one other person has described it on the Gitter recently. Seems to be related to the imauge packaged. The error appears after running the first cell (installing DLC and dependencies). But the cell finishes and I can proceed to the next several steps... but then the script hangs up at the Start Training Cell. See below for imauge error/ and hangup issue:

1) Occurs during Cell 1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
albumentations 0.1.12 requires imgaug<0.2.7,>=0.2.5, but you have imgaug 0.2.9 which is incompatible.


2) Occurs after running subsquent cells OK but hangs here at the Training Cell
Training parameters:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': '/content/drive/My Drive/runthrough_1-hartley-2021-08-04/dlc-models/iteration-0/runthrough_1Aug4-trainset95shuffle1/train/snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'adam', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 8, 'dataset_type': 'multi-animal-imgaug', 'deterministic': False, 'mirror': False, 'pairwise_huber_loss': False, 'weigh_only_present_joints': False, 'partaffinityfield_predict': True, 'pairwise_predict': True, 'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7], [8], [9], [10], [11], [12], [13], [14], [15], [16], [17], [18]], 'all_joints_names': ['tail_base', 'tail_tip', 'nose', 'L_ear', 'R_ear', 'R_paw', 'L_paw', 'R_foot', 'L_foot', 'body', 'R_hip', 'L_hip', 'R_shoulder', 'L_shoulder', 'R_brush_tip', 'L_brush_tip', 'R_brush_top', 'L_brush_top', 'brush_center'], 'alpha_r': 0.02, 'apply_prob': 0.5, 'clahe': True, 'claheratio': 0.1, 'crop_sampling': 'hybrid', 'crop_size': [400, 400], 'cropratio': 0.4, 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_runthrough_1Aug4/runthrough_1_hartley95shuffle1.pickle', 'decay_steps': 30000, 'display_iters': 500, 'edge': False, 'emboss': {'alpha': [0.0, 1.0], 'embossratio': 0.1, 'strength': [0.5, 1.5]}, 'histeq': True, 'histeqratio': 0.1, 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt', 'lr_init': 0.0005, 'max_input_size': 1500, 'max_shift': 0.4, 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_runthrough_1Aug4/Documentation_data-runthrough_1_95shuffle1.pickle', 'min_input_size': 64, 'multi_stage': True, 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]], 'net_type': 'resnet_50', 'num_idchannel': 0, 'num_joints': 19, 'num_limbs': 171, 'pafwidth': 20, 'pairwise_loss_weight': 0.1, 'partaffinityfield_graph': [[0, 1], [0, 2], [0, 3], [0, 4], [0, 5], [0, 6], [0, 7], [0, 8], [0, 9], [0, 10], [0, 11], [0, 12], [0, 13], [0, 14], [0, 15], [0, 16], [0, 17], [0, 18], [1, 2], [1, 3], [1, 4], [1, 5], [1, 6], [1, 7], [1, 8], [1, 9], [1, 10], [1, 11], [1, 12], [1, 13], [1, 14], [1, 15], [1, 16], [1, 17], [1, 18], [2, 3], [2, 4], [2, 5], [2, 6], [2, 7], [2, 8], [2, 9], [2, 10], [2, 11], [2, 12], [2, 13], [2, 14], [2, 15], [2, 16], [2, 17], [2, 18], [3, 4], [3, 5], [3, 6], [3, 7], [3, 8], [3, 9], [3, 10], [3, 11], [3, 12], [3, 13], [3, 14], [3, 15], [3, 16], [3, 17], [3, 18], [4, 5], [4, 6], [4, 7], [4, 8], [4, 9], [4, 10], [4, 11], [4, 12], [4, 13], [4, 14], [4, 15], [4, 16], [4, 17], [4, 18], [5, 6], [5, 7], [5, 8], [5, 9], [5, 10], [5, 11], [5, 12], [5, 13], [5, 14], [5, 15], [5, 16], [5, 17], [5, 18], [6, 7], [6, 8], [6, 9], [6, 10], [6, 11], [6, 12], [6, 13], [6, 14], [6, 15], [6, 16], [6, 17], [6, 18], [7, 8], [7, 9], [7, 10], [7, 11], [7, 12], [7, 13], [7, 14], [7, 15], [7, 16], [7, 17], [7, 18], [8, 9], [8, 10], [8, 11], [8, 12], [8, 13], [8, 14], [8, 15], [8, 16], [8, 17], [8, 18], [9, 10], [9, 11], [9, 12], [9, 13], [9, 14], [9, 15], [9, 16], [9, 17], [9, 18], [10, 11], [10, 12], [10, 13], [10, 14], [10, 15], [10, 16], [10, 17], [10, 18], [11, 12], [11, 13], [11, 14], [11, 15], [11, 16], [11, 17], [11, 18], [12, 13], [12, 14], [12, 15], [12, 16], [12, 17], [12, 18], [13, 14], [13, 15], [13, 16], [13, 17], [13, 18], [14, 15], [14, 16], [14, 17], [14, 18], [15, 16], [15, 17], [15, 18], [16, 17], [16, 18], [17, 18]], 'pos_dist_thresh': 17, 'pre_resize': [], 'project_path': '/content/drive/My Drive/runthrough_1-hartley-2021-08-04', 'rotation': 25, 'rotratio': 0.4, 'save_iters': 10000, 'scale_jitter_lo': 0.5, **'scale_jitter_up': 1.25, 'sharpen': False, 'sharpenratio': 0.3}
Starting multi-animal training....**

**Comment by Sai-Priyanka** [comment]
I used to have the same issue, it was because of the version incompatibility of deeplabcut and imgaug. deeplabcut==2.2 version installs imgaug==0.2.9 version library. With version "!pip install imgaug==0.4.0" before the training cell resolves the issue.

**Comment by roaldarbol** [comment]
@Sai-Priyanka That seems to solve it for me, thanks! (I actually just added it where all the other pip installs are).
I'm running the script right now, so I'll update this answer once it's done running.

**Comment by neurologic** [comment]
@Sai-Priyanka Thank you. I was being way too patient until finally searching for an issue with "AttributeError: 'int' object has no attribute 'nb_rows'" here. I did not do so for a while because although that error got listed, the code seemed to proceed to execute and said 'starting multi-animal training' without crashing. 
including "!pip install imgaug==0.4.0" solved it for me as well. I did see an error about incompatabilities regarding imgaug and tensorflow 1.15 versus 1.2 when I did this, but I did not keep track of the error and the training seems to be running so seems like not a fatal error. If it happens again I will include that detail.
